src/App.ts

```diff
@@ -6,84 +6,158 @@
+
   logger.info(
     `Server started at ${config.CONTROLLER.HOST}:${config.CONTROLLER.PORT}`
   );
   return server;
 }
 
 export function stopServer(server: http.Server): void {
-  logger.info("Stopping Proxy PagoPa Server...");
+  logger.info("Stopping Proxy PagoPA Server...");
   server.close();
 }
 
-// Set Restful WS routes
+/**
+ * Set RESTful WS endpoints
+ * @param {core.Express} app - Express server to set
+ * @param {Configuration} config - PagoPa proxy configuration
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {PagamentiTelematiciPspNodoAsyncClient} pagoPAClient - PagoPa SOAP client to call verificaRPT and attivaRPT services
+ */
 function setRestfulRoutes(
   app: core.Express,
   config: Configuration,
-  verificaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient,
-  attivaRPTPagoPaClient: pagoPaSoapClient.PagamentiTelematiciPspNodoAsyncClient
+  redisClient: redis.RedisClient,
+  pagoPAClient: pagoPASoapClient.PagamentiTelematiciPspNodoAsyncClient
 ): void {
+  const jsonParser = bodyParser.json();
   app.get(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_CHECK,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_REQUESTS_GET,
     (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Check Request (GET)...");
       toExpressHandler(
-        PaymentController.checkPaymentToPagoPa(
-          config.PAGOPA,
-          verificaRPTPagoPaClient
-        )
+        PaymentController.getPaymentInfo(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
   app.post(
-    config.CONTROLLER.ROUTES.RESTFUL.PAYMENTS_ACTIVATION,
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_POST,
+    jsonParser,
     (req: express.Request, res: express.Response) => {
-      logger.info("Serving Payment Activation Request (POST)...");
       toExpressHandler(
-        PaymentController.activatePaymentToPagoPa(
-          config.PAGOPA,
-          attivaRPTPagoPaClient
-        )
+        PaymentController.activatePayment(config.PAGOPA, pagoPAClient)
       )(req, res);
     }
   );
+  app.get(
+    config.CONTROLLER.ROUTES.RESTFUL.PAYMENT_ACTIVATIONS_GET,
+    (req: express.Request, res: express.Response) => {
+      toExpressHandler(PaymentController.getActivationStatus(redisClient))(
+        req,
+        res
+      );
+    }
+  );
+
+  // Endpoint for OpenAPI handler
+  app.get("/specs/api/v1/swagger.json", GetOpenapi(publicApiV1Specs));
+}
+
+/**
+ * Define and return a method to start a SOAP Server for PagoPA
+ * @param {redis.RedisClient} redisClient - The redis client used to store information sent by PagoPA
+ * @param {redisTimeout} redisTimeout - The timeout to use for information stored into redis
+ * @return {(app: core.Express) => soap.Server} A method to execute for start server listening
+ */
+function getSoapServer(
+  redisClient: redis.RedisClient,
+  redisTimeout: number
```

Comment:
name the parameter with the unit, is this in seconds or milliseconds? call it `redisTimeoutSecs` or `redisTimeoutMillis`